client/state/reader/streams/reducer.js

```diff
@@ -101,6 +101,10 @@ export const pendingItems = ( state = PENDING_ITEMS_DEFAULT, action ) => {
 			return { ...state, lastUpdated: maxDate };
 		case READER_STREAMS_UPDATES_RECEIVE:
 			streamItems = action.payload.streamItems;
+			if ( streamItems.length === 0 ) {
+				return state;
+			}
```

Comment:
This condition is already few lines down on line 118 -- I suppose it can be removed from there?

Reply:
No, that's actually a different check, as `streamItems` is mutated in between. This first one checks "did we get _any_ items?", while the second one checks "did we get any new items?".

It is a bit odd, though, so I'll refactor that next bit to remove the mutation and try and make it clearer.

Reply:
See #26036